consensus_encoding/src/encode/mod.rs

```diff
@@ -38,3 +38,24 @@ pub trait Encoder<'e> {
     /// and just call `current_chunk` to see if it works.
     fn advance(&mut self) -> bool;
 }
+
+/// Implements a newtype around an encoder which implements the
+/// [`Encoder`] trait by forwarding to the wrapped encoder.
+#[macro_export]
+macro_rules! encoder_newtype{
+    (
+        $(#[$($struct_attr:tt)*])*
+        pub struct $name:ident$(<$lt:lifetime>)?($encoder:ty);
```

Comment:
These macro interface quickly get gnarly for me, but I think the complexity is worth it here to capture the lifetime to weave it into the impl?